src/Workspaces/Remote/ServiceHub/Services/LanguageServer.cs

```diff
@@ -94,29 +94,28 @@ public Task<SymbolInformation[]> WorkspaceSymbolAsync(WorkspaceSymbolParams args
 
         private async Task<ImmutableArray<SymbolInformation>> SearchAsync(Solution solution, WorkspaceSymbolParams args, CancellationToken cancellationToken)
         {
-            // When progress reporting is supported, report incrementally per project and return an empty result at the end.
-            // Otherwise aggregate and return the results for all projects at the end.
+            // When progress reporting is supported, report incrementally per document and return an empty result at the end.
+            // Otherwise aggregate and return the results for all documents at the end.
             if (args.Progress != null)
             {
-                var tasks = solution.Projects.Select(p => SearchProjectAndReportSymbolsAsync(p, args, cancellationToken));
+                var tasks = solution.Projects.SelectMany(p => p.Documents).Select(d => SearchDocumentAndReportSymbolsAsync(d, args, cancellationToken));
```

Comment:
There may be a large increase in allocations here, for the number of tasks and closures, from n_projects to n_documents. This should be profiled with large solutions.

Reply:
even with large solutions, we're only in the thousands.  We also already parallelize over documents for several other sorts of operations.  It's never shown up as anything to be concerned about.  i.e. even 100k extra allocations here would be practically nothing.